### PR TITLE
[Feature][WIP] Report Privacy Related Capabilities in Admin

### PR DESCRIPTION
**Pull request by mbabker**
### Summary of Changes

This adds a reporting capability to `com_privacy` which allows us to show information to users about privacy related features and capabilities in core and installed extensions.  This screen is meant entirely as an informational dashboard to help site owners have awareness of what integrated extensions are doing.

screen shot 2018-05-16 at 9 29 00 pm

This is a read-only screen with append-only capabilities in plugins (meaning one plugin cannot remove the information reported by another and users do not have the ability to remove information in any way except disabling or uninstalling extensions).

### Testing Instructions

Install the base dev branch and apply this patch, or download this branch and install Joomla fresh.  You'll get the new capabilities screen in the component sidebar, navigate to it and you should see a screen with the sliders I have in the screenshot above.  Enabling the reCAPTCHA or language filter plugins should add additional sliders as they are also reporting capabilities.

### Documentation Changes Required

Yep, the plugin integration for this system

### TODO

- [ ] Verify the array merging syntax in the model is actually PHP 5.3 compatible (sorry, I don't have PHP 5 resources available to me anymore so I'm basically relying on CI, code linting, and user testing for this level of backward compatibility)
- [ ] Add reporting for the rest of core, this will be where privacy plugins start getting created

### CODE REVIEW

**Comment by JoomliC** [comment]
Sorry if i ask an already answered (and "stupid") question, but for 3rd party extensions, it supposes to create a privacy plugin for this task ? (in case of component/module)

About this PR, love the idea to centralize such information to help site owners to create their global Privacy Policy! 👍

**Comment by mbabker** [comment]
Yes extension developers will have to create privacy plugins to integrate to this system.

**Comment by JoomliC** [comment]
> Yes extension developers will have to create privacy plugins to integrate to this system.
Thanks!
I will test all this asap!

**Comment by manojLondhe** [comment]
Tested, seems to work fine. I even created privacy type of plugin for one of the extension and it shows info on the same screen.

privacy-capabilities

File: administrator/language/en-GB/en-GB.plg_system_languagefilter.ini [hunk 1]
```diff
@@ -24,5 +24,6 @@ PLG_SYSTEM_LANGUAGEFILTER_FIELD_REMOVE_DEFAULT_PREFIX_DESC="Remove the defined U
 PLG_SYSTEM_LANGUAGEFILTER_FIELD_REMOVE_DEFAULT_PREFIX_LABEL="Remove URL Language Code"
 PLG_SYSTEM_LANGUAGEFILTER_OPTION_SESSION="Session"
 PLG_SYSTEM_LANGUAGEFILTER_OPTION_YEAR="Year"
+PLG_SYSTEM_LANGUAGEFILTER_PRIVACY_CAPABILITY_LANGUAGE_COOKIE="On a site which supports multiple languages, this plugin can be configured to set a cookie on the user's browser which remembers their language preference. This cookie is used to redirect users to their preferred language when visiting the site and creating a new session."
```
Comment (manojLondhe) [inline administrator/language/en-GB/en-GB.plg_system_languagefilter.ini]: @mbabker do we need cookie name here too like added in cookie plugin?

Comment (mbabker) [reply administrator/language/en-GB/en-GB.plg_system_languagefilter.ini]: It's a hashed name so it's not clearly identifiable like the others.  That probably needs to be specified, but as the code is we can't give a clear "the name of this cookie (is|begins with) foo" statement.

Comment (manojLondhe) [reply administrator/language/en-GB/en-GB.plg_system_languagefilter.ini]: Yeah, I think we need to specify that.

**Comment by ot2sen** [comment]
Tested, works well. PHP 5.6/7 and various languages installed.
17-05-2018_privacy_capabilities

**Comment by mbabker** [comment]
Minus the voting plugin collecting IP addresses, everything in core is covered now.  I'll add that when plugins for core start getting built for the other system integrations.  Pending any issues I'll merge this tomorrow evening to move on to the next task.

**Comment by ggppdk** [comment]
Tested on PHP 7.2 and reviewed, works as described

**Comment by alikon** [comment]
do you think we need to add the plg_system_actionlogs for his capabilities to collect IP or is already included in the _Joomla Core Capabilities_

**Comment by mbabker** [comment]
It's already covered by core since it is explicitly stated that our logging system collects IPs.

**Pull request merged**